src/compiler/scala/tools/nsc/settings/Warnings.scala

```diff
@@ -24,6 +26,77 @@ trait Warnings {
   // Warning semantics.
   val fatalWarnings = BooleanSetting("-Werror", "Fail the compilation if there are any warnings.") withAbbreviation "-Xfatal-warnings"
 
+  private val WconfDefault = List("cat=deprecation:ws", "cat=feature:ws", "cat=optimizer:ws")
+  // Note: user-defined settings are added on the right, but the value is reversed before
+  // it's parsed, so that later defined settings take precedence.
+  val Wconf = MultiStringSetting(
+    "-Wconf",
+    "patterns",
+    "Configure reporting of compiler warnings; use `help` for details.",
+    helpText = Some(
+      s"""Configure compiler warnings.
+         |Syntax: -Wconf:<filters>:<action>,<filters>:<action>,...
+         |where <filters> is one or more <filter> combined with &, i.e., <filter>&...&<filter>
+         |
+         |Note: Run with `-Wconf:any:warning-verbose` to print warnings with their the category,
+         |site, origin (for deprecations) and since version (for deprecations).
```

Comment:
typo `their category`, suggest `print warnings with their category and site, and for deprecations, their origin and since version.`  What's origin? oh I guess the library name thing in the message, or the actual class? OK I see below it's the latter. You can say `since="mylib 1.0"` but I don't know if that is common usage (or used at all).

Reply:
> the library name thing in the message, or the actual class? OK I see below it's the latter

Yes, it's the full name of the deprecated symbol


> You can say `since="mylib 1.0"`

That is supported, in the sense that `mylib ` is dropped when parsing the `since` version. To filter deprecations for a specific library, use the `origin`, add a `since` filter if needed.